Node/src/TeamsMessage.ts

```diff
@@ -41,19 +41,151 @@ export enum MentionTextLocation {
+ *
+ */
+export class MentionEntity {
+  type: string;
+  mentioned: any;
+  text: string;
+}
+
+/**
+ * @class
+ * At mention user entity in message.
+ *
+ * @member {string} [type] at mention type, its value is always mention.
+ *
+ * @member {object} [mentioned] mentioned object with id, type and text value.
+ *
+ * @member {string} [text] text value to display in the message
+ *
+ */
+export class UserMention extends MentionEntity {
+  /**
+    *  Initialize a new instance of at mention user entity
+    *  @param {IIdentity} user - User object to at mention.
+    *  @param {string} text - At mention string to display.
+    */  
+  constructor(user: builder.IIdentity, text?: string) {
+    super();
+    if (!user || !user.id) {
+      throw new Error('Mentioned user and user ID cannot be null');
+    }
+
+    if (!user.name && !text) {
+      throw new Error('Either mentioned user name or mentionText must have a value');
+    }
+
+    let mentionEntityText = text || user.name;
+    this.type = 'mention';   
+    this.text = '<at>'+mentionEntityText+'</at>';
+    this.mentioned = {
+      'id' : user.id,
+      'name' : mentionEntityText,
+      'type': 'user'
+    };
+  }
+}
+
+/**
+ * @class
+ * At mention channel entity in message.
+ *
+ * @member {string} [type] at mention type, its value is always mention.
+ *
+ * @member {object} [mentioned] mentioned object with id, type and text value.
+ *
+ * @member {string} [text] text value to display in the message
+ *
+ */
+export class ChannelMention extends MentionEntity {
+  /**
+    *  Initialize a new instance of at mention channel entity
+    *  @param {ChannelInfo} channel - The channel to at mention.
+    */
+  constructor(channel: ChannelInfo) {
+    super();
+    if (!channel || !channel.id) {
+      throw new Error('Mentioned channel and channel ID cannot be null');
+    }
+
+    if (!channel.name) {
+      throw new Error('Channel name must have a value, use General as name if it is a team');
+    }
+
+    this.type = 'mention';
+    this.text = '<at>'+channel.name+'</at>';
+    this.mentioned = {
+      'id' : channel.id,
+      'name' : channel.name,
+      'type': 'channel'
+    };
+  }
+}
+
+/**
+ * @class
+ * At mention team entity in message.
+ *
+ * @member {string} [type] at mention type, its value is always mention.
+ *
+ * @member {object} [mentioned] mentioned object with id, type and text value.
+ *
+ * @member {string} [text] text value to display in the message
+ *
+ */
+export class TeamMention extends MentionEntity {
+  /**
+    *  Initialize a new instance of at mention team entity
+    *  @param {TeamInfo} team - The team to at mention.
+    */
+  constructor(team: TeamInfo) {
+    super();
+    if (!team || !team.id) {
+      throw new Error('Mentioned team and team ID cannot be null');
+    }
+
+    if (!team.name) {
+      throw new Error('Team name must have a value, use General as name if it is a team');
```

Comment:
We shouldn't use "general" as team name. "general" is the default channel name, never a team name.